FWCore/Framework/src/Worker.h

```diff
@@ -742,7 +860,22 @@ namespace edm {
       assert(*iEPtr);
       TransitionIDValue<typename T::MyPrincipal> idValue(ep);
       if(shouldRethrowException(*iEPtr, parentContext, T::isEvent_, idValue)) {
-        exceptionPtr = *iEPtr;
+        if (ranAcquireWithoutException_) {
```

Comment:
Any chance this could be done in the task wrapper? Else, maybe move to a different function to keep this template function from growing.

Reply:
I created a task wrapper to handle this and pushed a commit with that change. I could not figure out a way to get rid of the bool, but succeeded  in removing all changes from runAcquireAfterAsyncPrefetch. This part of the code is no longer modified.